### PR TITLE
Feature/dei 229 depth average select z or sigma from inputfile

### PR DESCRIPTION
**Pull request by mKlapwijk**

### CODE REVIEW

File: decoimpact/business/utils/dataset_utils.py [hunk 1]
```diff
@@ -6,9 +6,11 @@
 # https://github.com/Deltares/D-EcoImpact/blob/main/LICENSE.md
 """Library for utility functions regarding an xarray dataset"""
 
+from re import I
 from typing import List, Optional
 
 import xarray as _xr
+from numpy import var
```
Comment (CindyvdVries) [inline decoimpact/business/utils/dataset_utils.py:13]: are regex and numpy used in this file?

**Review by CindyvdVries — APPROVED** [review]

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 2]
```diff
@@ -109,20 +115,41 @@ def execute(
...
 
-        Returns:
-            values (List[str]): Values based on prefix + suffix name
-        """
-        variable = [value_arrays[name] for name in value_arrays if suffix in name][0]
-        return variable
+def _extract_variable_based_on_suffix(
+        value_arrays: Dict[str, _xr.DataArray],
+        suffix: str
+        ) -> list:
```
[the remark this answers is not in the document]
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py:122]: done

File: decoimpact/business/utils/dataset_utils.py (hunk 1 above)
Comment (mKlapwijk) [reply decoimpact/business/utils/dataset_utils.py:13]: removed

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 3]
```diff
@@ -148,8 +148,7 @@ def _get_layer_suffix(
...
     if layer_type.lower() == "sigma":
         return INTERFACES_SIGMA_SUFFIX
-    elif layer_type.lower() == "z":
+    if layer_type.lower() == "z":
         return INTERFACES_Z_SUFFIX
-    else:
-        logger.log_error(f"Layer type {layer_type} unknown. Interface "
-                            "variable could not be determined.")
+    logger.log_error(f"Layer type {layer_type} unknown. Interface "
+                     f"variable could not be determined.")
```
Comment (CindyvdVries) [inline decoimpact/business/entities/rules/depth_average_rule.py:154]: remove f -> no variable on this line
maybe add "Allowed layer type: "z" or "sigma"

**Review by CindyvdVries — APPROVED** [review]
looks good

File: decoimpact/business/entities/rules/depth_average_rule.py (hunk 3 above)
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py]: done

**Pull request merged**